test/functional/FunctionalCodeAction.hs

```diff
@@ -513,8 +518,14 @@ missingPragmaTests = testGroup "missing pragma warning code actions" [
             contents <- documentContents doc
 
             let expected =
+-- TODO: Why CPP???
+#if __GLASGOW_HASKELL__ < 810
+                    [ "{-# LANGUAGE ScopedTypeVariables #-}"
+                    , "{-# LANGUAGE TypeApplications #-}"
+#else
                     [ "{-# LANGUAGE TypeApplications #-}"
                     , "{-# LANGUAGE ScopedTypeVariables #-}"
+#endif
```

Comment:
Tests can pass now by adding this, though it's not clear to me...

Reply:
Well, we could force a ordering (alphabetical? i would say pre 8.10 is the right one)  but not sure if it worths tbh

Reply:
In this test case, `getParsedModule` should return `Nothing`, because the module will result in parse error without enabling TypeApplications. So the insert line is always `0`, which means TypeApplications will be inserted to the top (and inserting after the shebang will not work). But in tests under GHC 8.10, TypeApplications is inserted under ScopedTypeVariables, so It seems that in pre 8.10 cases `GetParsedModule`  rule does not return `Nothing`. 

Even if GHC parser returns `POk` instead of `PFailed` for non-fatal errors like this one since 8.10, `parseFileContents` should fail immediately when we encounter errors (if I understand correctly), so it's weird that GetParsedModule rule does not return `Nothing`...

Reply:
Weird indeed, i think #1340 failed to pass tests for the same reason

Reply:
Hmm actually #1340 failed to pass is expected. Test added by #1340 uses TypeApplications as the test object. As mentioned above, missing TypeApplications will result in parse error, so "after shebang" won't work in this case. #1340 should change another pragma to test and mark this case as an expected failure.

Reply:
oh, i see thanks for the insight
what would be the solution in that case (shebang + needed TypeApplications)? 
The plugin should suggest add it (to fix the parse error itself) but it would add it in the incorrect position (as we dont have the parsed module)

Maybe that combination is improbable though...

Reply:
We could use file contents as the fallback. But imo what's really rare is shebang -- inserting pragmas shouldn't have been depended on parsed module...

Reply:
@ishmum123 Please change another pragma to be tested with shebang.